yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryExecutor.java

```diff
@@ -203,46 +204,53 @@ private RelationalPreparedStatement prepareContinuationStatement(@Nonnull Relati
-                // Edge case: when there are no results at all, we don't want to store the result set at all,
-                // to create an empty aggregated result set
-                if (hasResult) {
-                    results.add(resultSet);
-                }
-                // Have continuations - keep running the query
-                Continuation continuation = resultSet.getContinuation();
-                while (!continuation.atEnd()) {
-                    try (var s2 = prepareContinuationStatement(connection, continuation, FORCED_MAX_ROWS)) {
-                        resultSet = (RelationalResultSet)executeStatement(s2, null);
-                        resultSet.next(); // Initialize result set value retrieval. Has only one row.
-                        continuation = resultSet.getContinuation();
-                        if (!continuation.atEnd()) {
-                            results.add(resultSet);
-                        }
+            return executeStatementWithForcedContinuations(s, queryString, connection);
+        } else {
+            if (maxRows != null) {
+                s.setMaxRows(maxRows);
+            }
+            // No change to behavior
+            return executeStatement(s, queryString);
+        }
+    }
+
+    private Object executeStatementWithForcedContinuations(final @Nonnull Statement s, final @Nullable String queryString, final RelationalConnection connection) throws SQLException {
+        s.setMaxRows(FORCED_MAX_ROWS);
+        Object result = executeStatement(s, queryString);
+        if (result instanceof RelationalResultSet) {
+            @SuppressWarnings("PMD.CloseResource")
+            RelationalResultSet resultSet = (RelationalResultSet)result;
+            List<RelationalResultSet> results = new ArrayList<>();
+            RelationalResultSetMetaData metadata = resultSet.getMetaData(); // The first metadata will be used for all
+
+            boolean hasResult = resultSet.next(); // Initialize result set value retrieval. Has only one row.
+            // Edge case: when there are no results at all, we don't want to store the result set at all,
+            // to create an empty aggregated result set
+            if (hasResult) {
+                results.add(resultSet);
+            }
+            // Have continuations - keep running the query
+            Continuation continuation = resultSet.getContinuation();
+            while (!continuation.atEnd()) {
+                try (var s2 = prepareContinuationStatement(connection, continuation, FORCED_MAX_ROWS)) {
+                    resultSet = (RelationalResultSet)executeStatement(s2, null);
+                    resultSet.next(); // Initialize result set value retrieval. Has only one row.
+                    continuation = resultSet.getContinuation();
+                    if (!continuation.atEnd()) {
+                        results.add(resultSet);
```

Comment:
IIUC if the total rows is an even multiple of the maxRows, the last resultSet is always an empty resultSet, and that's why this works.
But if that weren't the case, this would always be missing the last value.

Is it possible to have a ResultSet with a single row, and `getContinuation()` returns an end continuation? Such that we would drop that last row? We could have a bug where when maxrows=1 we always return the last row twice, which would be hidden by this flow.

Reply:
There are several reasons as to why this feature only works with `maxRows:1`, among them also the fact that the `AggregateResultSet` cannot advance to the last result of the result set in order to get a continuation without irrevocably changing the state of the result set. 
Added an assertion that would alleviate the above issue, but I think that we would need a more elaborate scheme to get this to work with maxRows!=1 (that was the original intent, but it turned more complex than originally thought)